### PR TITLE
Definir imagem padrão do thumbnail

### LINKED ISSUE #515: Definir imagem padrão do thumbnail
## Feature Description

Atualmente, as imagens são enviadas para a tabela de uploads, mas nenhuma delas é definida como `thumbnail_id`na tabela de tecnologias. Com isso, a informação de thumbnail não é carregada (mesmo quando uma requisição é feita para o backend incluindo o `embed`), impossibilitando de ser exibida nos cards. Essa definição deve ser feita no momento dos uploads. 

---------------

## Acceptance criteria

* 

## Implementation Brief

*

### PR DESCRIPTION
**Pull request by luizeboli**
## Summary

Resolves #515

## Relevant technical choices

* Tive que extrair a parte do preview das imagens pra um novo componente, pois não consegui integrar o grupo de radio buttons com o react-hook-form. Os inputs precisam ter unique names e pelo fato dos radios serem renderizados dinamicamente dentro do map, não consigo agrupá-los.

## QA Steps

Os passos de teste estão descritos no vídeo:

https://www.loom.com/share/6f28cafb064b413385309d513c5525a4?from_recorder=1

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code passes the linting.
- [x] My code has proper inline documentation.
- [x] I have manually tested this PR.

### CODE REVIEW

File: packages/web/components/TechnologyForm/MapAndAttachments/styles.js
Comment (mateus4k) [inline packages/web/components/TechnologyForm/MapAndAttachments/styles.js:73]: image

File: packages/web/components/TechnologyForm/MapAndAttachments/ImagesPreview.js
Comment (mateus4k) [inline packages/web/components/TechnologyForm/MapAndAttachments/ImagesPreview.js:40]: Try it:
```jsx
if (!previewedImgFiles.length) {
	return null; 
}

return (
	previewedImgFiles?.map() // [...]
); 
```

File: packages/web/components/Form/styles.js
Comment (mateus4k) [inline packages/web/components/Form/styles.js:42]:
```suggestion
	labelTop: css`
```

File: packages/web/components/Form/styles.js
Comment (mateus4k) [inline packages/web/components/Form/styles.js:80]: See suggestion above ☝️
```suggestion
		${labelPlacement === 'top' && inputWrapperModifiers.labelTop}
		${labelPlacement === 'right' && inputWrapperModifiers.labelRight}
```

**Review by mateus4k — CHANGES_REQUESTED** [review]
Awesome feature! 🚀

File: packages/web/components/TechnologyForm/MapAndAttachments/styles.js
Comment (luizeboli) [reply packages/web/components/TechnologyForm/MapAndAttachments/styles.js:73]: It was already there, I forgot to fix, ty 😄

File: packages/web/components/Form/styles.js
Comment (luizeboli) [reply packages/web/components/Form/styles.js:80]: It's a func because we can pass arguments, like theme or colors

File: packages/web/components/TechnologyForm/MapAndAttachments/ImagesPreview.js
Comment (lcnogueira) [inline packages/web/components/TechnologyForm/MapAndAttachments/ImagesPreview.js]:
```suggestion
					<label htmlFor={element.url}>
					<input
						id={element.url}
						type="radio"
						name="thumbnail_id"
						value={element.id}
						defaultChecked={value === element.id || !index}
						onChange={onChange}
					/>
						<FiImage fontSize="1.6rem" />
						Usar como capa
					</label>
```
In order to avoid the eslint message, you can wrap the input with the label tag 💡 .

File: packages/web/components/TechnologyForm/MapAndAttachments/styles.js
Comment (lcnogueira) [inline packages/web/components/TechnologyForm/MapAndAttachments/styles.js:157]: If you wrapp the input with the label tag, as suggested above, you'll need to update the style below:
```suggestion
		input {
			appearance: none;
		}
		input:checked + label {
			color: ${colors.white};
			background-color: ${colors.secondary};
			border: 2px solid ${colors.secondary};
		}
```

**Review by lcnogueira — APPROVED** [review]
Great job again, Felício 👏 . Looks great!

**Review by calaca — COMMENTED** [review]
Great job @luizeboli !

This feature is working just as intended.
image

image

I do think it's best to apply the suggestions made by @mateus4k so I'll wait for those to give approval.

File: packages/web/components/TechnologyForm/MapAndAttachments/styles.js
Comment (luizeboli) [reply packages/web/components/TechnologyForm/MapAndAttachments/styles.js:157]: Hey @lcnogueira, this `input:checked + label` won't work if I wrap input with label. Any thoughts?

Comment (luizeboli) [reply packages/web/components/TechnologyForm/MapAndAttachments/styles.js:157]: I could not achieve a good result by wrapping input 😞

Comment (luizeboli) [reply packages/web/components/TechnologyForm/MapAndAttachments/styles.js:157]: We have to style the whole label to fill it's background, otherwise it'll look like this:

image

Comment (luizeboli) [reply packages/web/components/TechnologyForm/MapAndAttachments/styles.js:157]: Any thoughts @calaca ?

Comment (calaca) [reply packages/web/components/TechnologyForm/MapAndAttachments/styles.js:157]: @luizeboli not really. I guess we should keep it as you did before, then.

I do have a visual suggestion:
image
Don't you think it looks better with all buttons with the same width as shown above the first picture?

Comment (luizeboli) [reply packages/web/components/TechnologyForm/MapAndAttachments/styles.js:157]: @calaca At first it looks slightly better, but on bigger breakpoints maybe it looks weird:

image

Comment (luizeboli) [reply packages/web/components/TechnologyForm/MapAndAttachments/styles.js:157]: I'll close this, ok? We can discuss it later as it's of such importance that @JoaoPauloSMoura also participate in these kind of discussions.

**Comment by luizeboli** [comment]
@calaca @lcnogueira @mateus4k changes applied 😄

File: packages/web/components/TechnologyForm/MapAndAttachments/styles.js
Comment (lcnogueira) [reply packages/web/components/TechnologyForm/MapAndAttachments/styles.js:157]: Agreed 👍

File: packages/web/components/TechnologyForm/MapAndAttachments/ImagesPreview.js
Comment (mateus4k) [inline packages/web/components/TechnologyForm/MapAndAttachments/ImagesPreview.js:26]: Did you see this comment? https://github.com/ufersa/plataforma-sabia/pull/558#discussion_r512611697 🤔

**Review by mateus4k — COMMENTED** [review]
I found a little problem during QA testing. I will explain it to you privately 👍

File: packages/web/components/TechnologyForm/MapAndAttachments/ImagesPreview.js
Comment (luizeboli) [reply packages/web/components/TechnologyForm/MapAndAttachments/ImagesPreview.js:26]: Yes bro no way to achieve this style doing that. Ive explained it in the second comment after that one youve mentioned

Comment (mateus4k) [reply packages/web/components/TechnologyForm/MapAndAttachments/ImagesPreview.js:26]: Sorry bro 👀

Comment (luizeboli) [reply packages/web/components/TechnologyForm/MapAndAttachments/ImagesPreview.js:26]: No problem hahaha take it easy

**Review by calaca — APPROVED** [review]
Approved! :100:

**Comment by luizeboli** [comment]
@mateus4k I'll open another issue to fix this problem as this isn't related to this one 🤝

**Review by mateus4k — APPROVED** [review]

**Review by mateus4k — APPROVED** [review]

**Pull request merged**
